packages/stencil/components/pagination/src/components/osds-pagination/osds-pagination.tsx

```diff
@@ -0,0 +1,200 @@
+
+  /** @see OdsPaginationAttributes.defaultCurrent */
+  @Prop({ reflect: true }) defaultCurrent?: number = odsPaginationDefaultAttributes.defaultCurrent;
+
+  /** @see OdsPaginationAttributes.totalPages */
+  @Prop({ reflect: true }) totalPages: number = odsPaginationDefaultAttributes.totalPages;
+
+  /** @see OdsPaginationAttributes.disabled */
+  @Prop({ reflect: true, mutable: true }) disabled: boolean = odsPaginationDefaultAttributes.disabled;
+
+  /** @see OdsPaginationEvents.odsCurrentChange */
+  @Event() odsCurrentChange!: EventEmitter<OdsPaginationCurrentChangeEventDetail>;
+
+  @Watch('defaultCurrent')
+  onDefaultCurrentChange(defaultCurrent?: number) {
+    this.logger.debug(`current: ${this.current}]`, 'defaultCurrent: ', defaultCurrent);
+  }
+
+  componentWillLoad() {
+    this.onDefaultCurrentChange();
+
+    if (this.defaultCurrent !== undefined) {
+      this.current = this.defaultCurrent;
+    }
+
+    this.current = this.current;
+  }
+
+  async componentDidUpdate() {
+    (this.el.shadowRoot?.querySelector('.selectedpage') as HTMLStencilElement)?.focus();
+  }
+
+  @Watch('current')
+  async onCurrentChange(current: number, oldCurrent?: number) {
+    this.logger.log(`current: ${this.current}]`, 'current changed. emit new current', { current });
+    this.emitChange(current, oldCurrent);
+  }
+
+  private emitChange(current: number, oldCurrent?: number) {
+    this.logger.debug('emit', { current, oldCurrent });
+    this.odsCurrentChange.emit({
+      current: current,
+      oldCurrent: oldCurrent,
+    });
+  }
+
+  /**
+   * @internal
+   * @see OdsPaginationMethods.setPageIndex
+   */
+  @Method()
+  async setPageIndex(current: number) {
+    this.current = current;
+  }
+
+  onKeyPress = (event: any, page: any) => {
+    if (event.keyCode === 13 || event.keyCode === 32) {
+      this.setPageIndex(page);
+    }
+  };
+
+  render() {
+    const { totalPages, disabled } = this;
+
+    let pageList = this.controller.createPageList(totalPages, this.current);
+
+    return (
+      <Host
+        {...{
+          class: disabled ? 'disabled' : '',
+        }}
+      >
+        <div>
+          <ul>
+            <li class="arrows">
+              <osds-button
+                variant={OdsButtonVariant.ghost}
+                color={OdsThemeColorIntent.primary}
+                disabled={disabled ? true : this.current == 1}
+                onKeyDown={(event: any) => {
+                  if (this.current > 1 && !disabled) this.onKeyPress(event, Number(this.current) - 1);
+                }}
+                onClick={() => {
+                  this.setPageIndex(Number(this.current) - 1);
+                }}
+                size={OdsButtonSize.sm}
+              >
+                <osds-icon size={OdsIconSize.sm} name={OdsIconName.CHEVRON_LEFT} color={OdsThemeColorIntent.primary} class={this.current == 1 ? 'disabled' : ''}></osds-icon>
```

Comment:
instead setting `disabled` class here, you could set `disabled` on the parent button no ?